### ISSUE TITLE
2nd row of thumbnails not displaying Safari MacOS

### ISSUE THREAD

**Issue by jdogskizzle** [ajayyy/DeArrow#206]
<img width="1100" alt="Screenshot 2023-12-20 at 12 02 53 PM" src="https://github.com/ajayyy/DeArrow/assets/66547421/a056ecdd-acda-4f23-9aa7-90cd70f1909b">

As shown in the image, thumbnails will not properly load. This is fixed by refreshing the page, but on initial load this will happen.

**Comment by ajayyy** [comment]
What version of safari and DeArrow, and what other extensions do you have installed?

**Comment by ajayyy** [comment]
Can you try uninstalling and reinstalling?

**Comment by jdogskizzle** [comment]
Safari 17.1.2, DeArrow 1.2.24, 
<img width="205" alt="Screenshot 2023-12-20 at 1 58 14 PM" src="https://github.com/ajayyy/DeArrow/assets/66547421/144b7afd-9dfd-461b-ace2-6912977c8392">

Here's the extensions I currently have (All of the Adguard ones are checked)

**Comment by ajayyy** [comment]
Looks similar to https://github.com/ajayyy/DeArrow/issues/137#issuecomment-1704390085

**Comment by ajayyy** [comment]
Can you try uninstalling and reinstalling? And what happens if you disable the SponsorBlock extension?

**Comment by jdogskizzle** [comment]
Neither of those worked

**Comment by ajayyy** [comment]
Do you have any channel overrides?

**Comment by jdogskizzle** [comment]
no

**Comment by ajayyy** [comment]
Can you go to options -> misc -> copy debug information

**Comment by jdogskizzle** [comment]
{
    "debug": {
        "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.1.2 Safari/605.1.15",
        "platform": "MacIntel",
        "language": "en-US",
        "extensionVersion": "1.2.24"
    },
    "config": {
        "titleReplacements": 599,
        "channelOverrides": {},
        "vip": false,
        "alreadyActivated": true,
        "titleFormatting": 2,
        "darkMode": true,
        "thumbnailReplacements": 2568,
        "customConfigurations": {},
        "showInfoAboutRandomThumbnails": true,
        "alwaysShowShowOriginalButton": true,
        "allowExpirements": true,
        "showDonationLink": true,
        "showUpsells": true,
        "donateClicked": 0,
        "invidiousInstances": 0,
        "keepUnsubmitted": true,
        "keepUnsubmittedInPrivate": false,
        "shouldCleanEmojis": true,
        "serverAddress": "Default server address",
        "thumbnailServerAddress": "https://dearrow-thumb.ajay.app",
        "fetchTimeout": 7000,
        "startLocalRenderTimeout": 2000,
        "renderTimeout": 25000,
        "thumbnailCacheUse": 2,
        "showGuidelineHelp": true,
        "thumbnailFallback": 0,
        "thumbnailFallbackAutogenerated": 0,
        "extensionEnabled": true,
        "defaultToCustom": true,
        "showOriginalOnHover": false,
        "importedConfig": false,
        "replaceTitles": true,
        "replaceThumbnails": true,
        "useCrowdsourcedTitles": true,
        "titleMaxLines": 3,
        "showIconForFormattedTitles": true,
        "countReplacements": true,
        "ignoreAbThumbnails": true,
        "activated": true,
        "freeAccessWaitingPeriod": 259200000,
        "firefoxOldContentScriptRegistration": false,
        "lastIncognitoStatus": false,
        "showActivatedMessage": false
    }
}

**Comment by ajayyy** [comment]
Are you using the Safari version through the app store or your own build?

**Comment by jdogskizzle** [comment]
I am using stock safari

**Comment by ajayyy** [comment]
I mean the DeArrow extension, is it from https://apps.apple.com/us/app/dearrow-for-youtube/id6451469297

**Comment by jdogskizzle** [comment]
Yes the one from the App Store

**Comment by ajayyy** [comment]
Has this issue always occurred, or did it start occurring later after installing?

**Comment by ajayyy** [comment]
Can you send a screen recording of opening youtube.com, waiting a few seconds, then clicking a video, waiting a few seconds, then clicking the back button then scroll down until more thumbnails load on the right

**Comment by jdogskizzle** [comment]
[2023-12-24 14-23-43.mkv.zip](https://github.com/ajayyy/DeArrow/files/13762946/2023-12-24.14-23-43.mkv.zip)

**Comment by ajayyy** [comment]
thanks a lot! Could you open youtube.com, then open inspect element by right clicking and clicking inspect element. Then clickt the console tab and type the following into the console and send the response?

```js
console.log(window.location.host, window, window?.customElements, window?.customElements?.define, window.customElements?.define?.toString());
console.log(window["versionCB"], document.querySelector("#sponsorblock-document-script"), document.querySelectorAll("#sponsorblock-document-script"))
console.log(window.fetch)
```

**Comment by ajayyy** [comment]
Also, could you tell me what userscripts you have installed with your userscript manager?

**Comment by jdogskizzle** [comment]
<img width="1597" alt="Screenshot 2023-12-24 at 11 35 39 PM" src="https://github.com/ajayyy/DeArrow/assets/66547421/eefac8f8-fe3a-4236-8118-cd9434b75255">

<img width="798" alt="Screenshot 2023-12-24 at 11 36 04 PM" src="https://github.com/ajayyy/DeArrow/assets/66547421/f5ae3682-0e2b-4d0f-a7d2-004cdad267ea">

**Comment by ajayyy** [comment]
Could you run the following commands in the console:

```js
window.addEventListener("message", console.log)
document.createElement("ytd-thumbnail")
console.log("finished setup")
```

Then open youtube.com, wait a few seconds, then click a video, wait a few seconds, then click the back button then scroll down until more thumbnails load on the right

Then send the result from the console.

***

Could you also try disabling all other Safari extensions and see if the issue still occurs?

Thanks

**Comment by caveman1973** [comment]
I had the same problem. Enabling the extensions in Private Mode fixed the problem.

**Comment by mchungy** [comment]
I have the same problem too here, it seems like the banners are not loading with Sponsorship Block.

**Comment by MRuy** [comment]
I have the same problem but disabling the extension BlockTube fixed it

**Comment by ajayyy** [comment]
Issue with BlockTube fixed in 1.5.3
